source/geod24/LocalRest.d

```diff
@@ -410,6 +331,61 @@ public final class RemoteAPI (API, alias S = VibeJSONSerializer!()) : API
 {
     static assert (!serializerInvalidReason!(S).length, serializerInvalidReason!S);
 
+    alias RestChannel = C.Channel!Variant;
+
+    /// Ask the node to shut down
+    private struct ShutdownCommand (API)
+    {
+        /// Any callback to call before the Node's destructor is called
+        void function (API) callback;
+
+        /// Whether we're restarting or really shutting down
+        bool restart;
+    }
+
+    /// Data sent by the caller
+    private struct Command
+    {
+        /// Channel to be used when responding to this Command
+        RestChannel resp_chn;
+        /// Method to call
+        string method;
+        /// Serialized arguments to the method
+        SerializedData args;
+    }
+
+    // very simple & limited variant, to keep it performant.
+    // should be replaced by a real Variant later
+    static struct Variant
+    {
+        this (Command msg) { this.cmd = msg; this.tag = Variant.Type.command; }
+        this (Response msg) { this.res = msg; this.tag = Variant.Type.response; }
+        this (FilterAPI msg) { this.filter = msg; this.tag = Variant.Type.filter; }
+        this (TimeCommand msg) { this.time = msg; this.tag = Variant.Type.time_command; }
+        this (ShutdownCommand!API msg) { this.shutdown = msg; this.tag = Variant.Type.shutdown_command; }
+
+        union
+        {
+            Command cmd;
+            Response res;
+            FilterAPI filter;
+            TimeCommand time;
+            ShutdownCommand!API shutdown;
+        }
+
+        ubyte tag;
```

Comment:
If you define an `enum`, why is this not typed ?

Reply:
It must have carried over from the old code. Will fix.